build_all/make_sdk.py

```diff
@@ -0,0 +1,192 @@
+
+    return usage_txt
+
+def parse_opts():
+    options, remainder = getopt.gnu_getopt(sys.argv[1:], 'ho:d:', ['output', 'help', 'device'])
+    # print('OPTIONS   :', options)
+    for opt, arg in options:
+        if opt in ('-h', '--help'):
+            print(usage())
+        elif opt in ('-o', '--output'):
+            commands_dict.output_path = arg
+            
+
+def run():
+    # set up paths for copying
+    output_path = os.path.abspath(commands_dict.output_path)
+    arduino_repo_root = os.path.abspath('../')
+    print(output_path)
+    arduino_pal_path = arduino_repo_root+'/pal/'
+    azure_iot_sdk_path = arduino_repo_root+'/sdk/'
+    AzureIoTHub_path = output_path+'/AzureIoTHub/'
+    AzureIoTProtocolHTTP_path = output_path+'/AzureIoTProtocol_HTTP/'
+    AzureIoTProtocolMQTT_path = output_path+'/AzureIoTProtocol_MQTT/'
+    AzureIoTUtility_path = output_path+'/AzureIoTUtility/'
+    AzureIoTSocketWiFi_path = output_path+'/AzureIoTSocket_WiFi/'
+    # AzureIoTSocketEthernet_path = output_path+'/AzureIoTSocket_Ethernet/'
+    AzureUHTTP_path = AzureIoTProtocolHTTP_path+'src/azure_uhttp_c/'
+    AzureUMQTT_path = AzureIoTProtocolMQTT_path+'src/azure_umqtt_c/'
+    SharedUtility_path = AzureIoTUtility_path+'src/azure_c_shared_utility/'
+    Adapters_path = AzureIoTUtility_path+'src/adapters/'
+    Macro_Utils_path = AzureIoTUtility_path+'src/azure_c_shared_utility/azure_macro_utils/'
+    Hub_Macro_Utils_path = AzureIoTHub_path+'src/azure_macro_utils/'
+    Umock_c_path = AzureIoTUtility_path+'src/umock_c/'
+    sdk_path = AzureIoTHub_path+'src/'
+    internal_path = AzureIoTHub_path+'src/internal/'
+
+    if (os.path.exists(output_path)):
+        #clear it out
+        pattern_delete_folder(output_path+'/Azure*')
+    else:
+        os.mkdir(output_path)
+
+    dir_util.copy_tree(arduino_repo_root+'/build_all/base-libraries/AzureIoTHub', AzureIoTHub_path)
+    dir_util.copy_tree(arduino_repo_root+'/build_all/base-libraries/AzureIoTUtility', AzureIoTUtility_path)
+    dir_util.copy_tree(arduino_repo_root+'/build_all/base-libraries/AzureIoTProtocol_HTTP', AzureIoTProtocolHTTP_path)
+    dir_util.copy_tree(arduino_repo_root+'/build_all/base-libraries/AzureIoTProtocol_MQTT', AzureIoTProtocolMQTT_path)
+    dir_util.copy_tree(arduino_repo_root+'/build_all/base-libraries/AzureIoTSocket_WiFi', AzureIoTSocketWiFi_path)
+    # dir_util.copy_tree(arduino_repo_root+'/build_all/base-libraries/AzureIoTSocket_Ethernet2', AzureIoTSocketEthernet_path)
+
+    shutil.copy2(azure_iot_sdk_path+'LICENSE', AzureIoTHub_path+'LICENSE')
+
+    dir_util.copy_tree(azure_iot_sdk_path+'iothub_client/src/', sdk_path)
+    dir_util.copy_tree(azure_iot_sdk_path+'iothub_client/inc/', sdk_path)
+    dir_util.copy_tree(azure_iot_sdk_path+'iothub_client/inc/internal/', internal_path)
+    dir_util.copy_tree(azure_iot_sdk_path+'serializer/src/', sdk_path)
+    dir_util.copy_tree(azure_iot_sdk_path+'serializer/inc/', sdk_path)
```

Comment:
is it possible to break this into sections?


Sections:

// Copy all sdk includes

// copy all sdk src

// copy certs

// copy samples

// Copy pal src

//copy Licenses 

This way code will be more organized and easy to modify in future.

Reply:
Did something similar.

Reply:
I did something similar.